src/main/java/org/cryptomator/windows/keychain/WindowsProtectedKeychainAccess.java

```diff
@@ -92,16 +94,16 @@ public String displayName() {
 	}
 
 	@Override
-	public void storePassphrase(String key, String displayName, CharSequence passphrase, boolean ignored) throws KeychainAccessException {
+	public void storePassphrase(String key, String displayName, CharSequence passphrase, boolean requireAuth) throws KeychainAccessException {
 		loadKeychainEntriesIfNeeded();
 		ByteBuffer buf = UTF_8.encode(CharBuffer.wrap(passphrase));
 		byte[] cleartext = new byte[buf.remaining()];
 		buf.get(cleartext);
 		var salt = generateSalt();
-		var ciphertext = dataProtection.protect(cleartext, salt);
+		var ciphertext = requireAuth ? windowsHello.setEncryptionKey(cleartext, salt) : dataProtection.protect(cleartext, salt);
 		Arrays.fill(buf.array(), (byte) 0x00);
 		Arrays.fill(cleartext, (byte) 0x00);
-		keychainEntries.put(key, new KeychainEntry(ciphertext, salt));
+		keychainEntries.put(key, new KeychainEntry(ciphertext, salt, requireAuth));
```

Comment:
This would alter the program flow, as this is a library and an KeychainAccessException needs to bubble up to the application using the library.

I'll keep the code as it is.